packages/application/src/shell.ts

```diff
@@ -657,6 +664,25 @@ class ApplicationShell extends Widget {
     }
     this._currentChanged.emit(args);
     this._onLayoutModified();
+
+    this._onCurrentSession(args.newValue ? (args.newValue as any).session : undefined);
+  }
+
+  /**
+   * Handle a change to the session of the current widget.
+   */
+  private _onCurrentSession(session?: IClientSession) {
```

Comment:
Can we change these to `_onCurrentSessionChanged` and `_onCurrentStatusChanged`, or similar? I found the names here to be kind of confusing.

Reply:
Yep, sounds good, will do.